src/components/AttachmentCarousel.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import {Pressable, View} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import _ from 'lodash';
+import styles from '../styles/styles';
+import * as Expensicons from './Icon/Expensicons';
+import Icon from './Icon';
+import reportActionPropTypes from '../pages/home/report/reportActionPropTypes';
+import CONFIG from '../CONFIG';
+import CONST from '../CONST';
+import ONYXKEYS from '../ONYXKEYS';
+
+const propTypes = {
+    /** sourceUrl to determine the initial index of the attachment rendered in it's respective actions */
+    sourceURL: PropTypes.string,
+
+    /** Object of report actions for this report */
+    reportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
+
+    /** callback from the parent to change the name and sourceUrl of parent's state */
+    onArrowPress: PropTypes.func,
+};
```

Comment:
Can you add meaningful comments to `sourceURL` & `onArrowPress` without being too technical